guides/source/action_controller_overview.md

```diff
@@ -899,6 +899,61 @@ Rails collects all of the parameters sent along with the request in the `params`
 [`query_parameters`]: https://api.rubyonrails.org/classes/ActionDispatch/Request.html#method-i-query_parameters
 [`request_parameters`]: https://api.rubyonrails.org/classes/ActionDispatch/Request.html#method-i-request_parameters
 
+#### `request.variant`
+
+Controllers might need to tailor a response based on context-specific information in a request. For example, controllers responding to requests from a mobile platform might need to render a HTML, JSON, or XML different content than requests from a desktop browser. They can add that context to the requests variant so that views render appropriately. Variant names are arbitrary, and can communicate anything from the request's platform (`:anrdoid`, `:ios`, `:linux`, `:macos`, `:windows`) to its browser (`:chrome`, `:edge`, `:firefox`, `:safari`), to the type of user (`:admin`, `:guest`, `:user`).
```

Comment:
Personally, I've only ever really used variants for platform hints, but felt that was too prescriptive. 

Are there other use cases worth highlighting, or would limiting this suggestion to platforms be best?

Reply:
Responsive design and CSS breakpoints have made the `:tablet`, `:phone`, and `:desktop` suggestion a little outdated.

Reply:
Not sure if we would like to add this use case but I found variant to be very helpful when implementing a new design or in combination with a feature flag when the feature is a major revamp of a view. One of the advantage of using variants it that from an I18n point of view the path remains the same so we don't broke the translation lazy keys.